contracts/MarginCalculator.sol

```diff
@@ -99,17 +370,146 @@ contract MarginCalculator {
         return cashValueInCollateral.toScaledUint(collateralDecimals, true);
     }
 
+    // structs to avoid stack too deep error
+    // struct to store shortAmount, shortStrike and shortUnderlyingPrice scaled to 1e27
+    struct ShortScaledDetails {
+        FPI.FixedPointInt shortAmount;
+        FPI.FixedPointInt shortStrike;
+        FPI.FixedPointInt shortUnderlyingPrice;
+    }
+    // struct to store isLiquidatable and the price for 1 otoken repaid scaled to the collateral decimals
+    struct LiquidationStatus {
+        bool isLiquidatable;
+        uint256 debtPrice;
+    }
+
+    /**
+     * @notice check if a specific vault is undercollateralized at a specific chainlink round
+     * @dev if the vault is of type 0, the function will revert
+     * @param _vault vault struct
+     * @param _vaultType vault type (0 for max loss/spread and 1 for naked margin vault)
+     * @param _vaultLatestUpdate vault latest update (timestamp when latest vault state change happened)
+     * @param _roundId chainlink round id
+     * @return isLiquidatable, true if vault is undercollateralized, liquidation price and collateral dust amount
+     */
+    function isLiquidatable(
+        MarginVault.Vault memory _vault,
+        uint256 _vaultType,
+        uint256 _vaultLatestUpdate,
+        uint256 _roundId
+    )
+        external
+        view
+        returns (
+            bool,
+            uint256,
+            uint256
+        )
+    {
+        // create LiquidationStatus struct to avoid stack too deep error
+        LiquidationStatus memory liquidationStatus = LiquidationStatus(false, 0);
+
+        // liquidation is only supported for naked margin vault
+        require(_vaultType == 1, "MarginCalculator: invalid vault type to liquidate");
```

Comment:
Don't have strong opinions about reverting this view function - but should discuss how we will program a liquidation bot to provide to community and run ourselves and if reverts on view functions will be unexpected or cause issues.

I understand the rationale for providing something other than false when a vault can never be liquidated vs just not liquidatable now.

Reply:
It make sense to use the revert for input validation, and false when all input are correct but the vault is not underwater.